Realm/Realm.Weaver/Extensions/TypeReferenceExtensions.cs

```diff
@@ -46,29 +46,32 @@ SequencePoint GetPropSequencePoint()
         }
     }
 
-    public static bool IsRealmObjectDescendant(this TypeReference @this, ImportedReferences references)
+    private static bool IsTObjectDescendant(TypeReference target, params TypeReference[] instanceOf)
```

Comment:
```suggestion
    private static bool IsDescendantOf(TypeReference @this, params TypeReference[] targetTypes)
```

I got a little confused as to what `target` and `instanceOf` mean. I'm not sure my suggestion is much better, but I think it's a step in the right direction.